server/src/main/java/org/opensearch/action/InputValidator.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action;
+
+public interface InputValidator {
```

Comment:
I feel like something like this must already exist somewhere... seems like a general problem that many APIs would have to solve